scripts/updates/check-updates.sh

```diff
@@ -10,85 +10,115 @@ BASEDIR=$(dirname "$(realpath "$0")")
+	if [ "${auto_update_flag}" != "true" ]; then
+		continue
+	fi
 
-		# Latest version is the current release tag on Github.
-		latest_version=$(curl --silent --location -H "Authorization: token ${GITHUB_API_TOKEN}" "https://api.github.com/repos/${project}/releases/latest" | jq -r .tag_name)
+	# Extract github project from TERMUX_PKG_SRCURL
+	project="$(echo "${srcurl}" | grep github.com | cut -d / -f4-5)"
+	if [ -z "${project}" ]; then
+		echo "Package ${package}'s TERMUX_PKG_SRCURL=${srcurl} is not a github archive url."
+		exit 1
+	fi
 
-		# Remove leading 'v' which is common in version tag.
-		latest_version=${latest_version#v}
+	# Our local version of package.
+	termux_epoch="$(echo "$termux_version" | cut -d: -f1)"
+	termux_version=$(echo "$termux_version" | cut -d: -f2-)
+	if [ "$termux_version" == "$termux_epoch" ]; then
+		# No epoch set.
+		termux_epoch=""
+	else
+		termux_epoch+=":"
+	fi
 
-		# If needed, filter version numbers from tag by using regexp.
-		if [ -n "$version_regexp" ]; then
-			latest_version=$(grep -oP "$version_regexp" <<< "$latest_version" || true)
-		fi
+	# Latest version is the current release tag on Github.
+	latest_version=$(curl --silent --location -H "Authorization: token ${GITHUB_API_TOKEN}" "https://api.github.com/repos/${project}/releases/latest" | jq -r .tag_name)
 
-		if [[ -z "$latest_version" || "$latest_version" = "null" ]]; then
-			echo "Failed to get latest version for '${package}'. Update 'github-projects.txt'."
-			exit 1
-		fi
+	# Remove leading 'v' which is common in version tag.
+	latest_version=${latest_version#v}
 
-		# Translate "_" into ".".
-		latest_version=${latest_version//_/.}
+	# If the github api returns error
+	if [ -z "$latest_version" ] || [ "${latest_version}" = "null" ]; then
+		echo "Failed to get latest version for '${package}'. Check whether the release of repo https://github.com/${project} it is published."
+		exit 2
+	fi
 
-		# We have no better choice for comparing versions.
-		if [ "$(echo -e "${termux_version}\n${latest_version}" | sort -V | head -n 1)" != "$latest_version" ] ;then
-			if [ "$BUILD_PACKAGES" = "false" ]; then
-				echo "Package '${package}' needs update to '${latest_version}'."
-			else
-				echo "Updating '${package}' to '${latest_version}'."
-				sed -i "s/^\(TERMUX_PKG_VERSION=\)\(.*\)\$/\1${termux_epoch}${latest_version}/g" "${BASEDIR}/../../packages/${package}/build.sh"
-				sed -i "/TERMUX_PKG_REVISION=/d" "${BASEDIR}/../../packages/${package}/build.sh"
-				echo n | "${BASEDIR}/../bin/update-checksum" "$package" || {
-					echo "Failed to update checksum for '${package}', skipping..."
-					git checkout -- "${BASEDIR}/../../packages/${package}"
+	# Translate "_" into ".".
+	latest_version=${latest_version//_/.}
+
+	latest_version_raw=$latest_version
+	# If needed, filter version numbers from tag by using regexp.
+	if [ -n "$version_regexp" ]; then
+		latest_version=$(grep -oP "$version_regexp" <<< "$latest_version" || true)
+	fi
```

Comment:
You have messed up the order by which version tags are processed. All translations except stripping the leading `v` should be done after regexp filter.

Anyway, I'm fixing script on my own now as well as test it.